docs/references/architecture/adr-113-mempool-lanes.md

```diff
@@ -53,32 +55,43 @@ is an orthogonal problem to transaction classification, although the latter may
 
 Before jumping into the design of the proposal, we define more formally the properties supported by
 the current implementation of the mempool. Then we state what properties the new mempool should
-offer to guarantee the desired QoS.
+offer to guarantee the desired QoS. The following definitions are common to all properties.
 
-The following definition is common to all properties.
+When attempting to add an incoming transaction to the mempool, the node first checks that it is not
+already in the cache before checking its validity with the application. 
 
-:memo: _Definition_: Given any two different transactions `tx1` and `tx2`, we say that `tx1` is
-*processed and disseminated before* `tx2` in a given node, when:
-- `tx1` is reaped from the mempool to form a block proposal before `tx2`,
-- `tx1` is rechecked before `tx2`, and
-- `tx1` is sent to a given peer before `tx2`.
+:memo: _Definition_: We say that a node receives a transaction `tx` _for the first time_ when the
+node receives `tx` and `tx` is not in the cache.
 
-Note that in the current implementation there is one dissemination routine per peer, so it could
-happen that `tx2` is sent to a peer before `tx1` is sent to a different peer.
+By this definition, it is possible that a node receives a transaction "for the first time", then
+gets the transaction evicted from the cache, and at a later time receives it "for the first time"
+again. The cache implements a Least-Recently Used (LRU) policy for removing entries when the
+cache is full.
+
+:memo: _Definition_: Given any two different transactions `tx1` and `tx2`, in a given node, we say that:
+1. `tx1` is *validated before* `tx2`, when `tx1` and `tx2` are received for the first time, and `tx1`
+is validated against the application (via `CheckTx`) before `tx2`,
+1. `tx1` is *rechecked before* `tx2`, when `tx` and `tx2` are in the mempool and `tx1` is
+re-validated (rechecked via `CheckTx`) before `tx2`,
```

Comment:
What if `tx1` and `tx2` are re-checked in different heights? Does that case fit into this definition, or we're leaving such a case out on purpose?

Reply:
If both transactions are in the mempool, that means that they are rechecked at the same height. I'll clarify in the text.